sdk/python/tests/integration/online_store/test_universal_online.py

```diff
@@ -1012,29 +1004,30 @@ def test_go_server_life_cycle(go_cycle_environment, go_data_sources):
         go_cycle_environment.start_date - timedelta(days=1),
         go_cycle_environment.end_date + timedelta(days=1),
     )
-    expected_values = df.sort_values(by="driver_id")
     features = [f"{simple_driver_fv.name}:value"]
     entity_rows = [{"driver": driver_id} for driver_id in sorted(driver_entities)]
 
     # Start go server process that calls get_online_features and return and check if at any time go server
     # fails to clean up resources
-    import os
-    import signal
-
-    # Duplicate the current test suit in the child process
-    child_pid = os.fork()
-    if child_pid == 0:
-        online_features = fs.get_online_features(
-            features=features, entity_rows=entity_rows
-        ).to_dict()
-        assert np.allclose(expected_values["value"], online_features["value"])
-        os.kill(os.getpid(), signal.SIGTERM)
-        os._exit(0)
-    os.wait()
+    fs.get_online_features(features=features, entity_rows=entity_rows).to_dict()
+
+    assert (
+        fs._go_server
+        and fs._go_server._go_server_started.is_set()
+        and fs._go_server._shared_connection._process
+    )
+    go_fs_pid = fs._go_server._shared_connection._process.pid
+
+    os.kill(go_fs_pid, signal.SIGTERM)
     # At the same time checking that resources are clean up properly once child process is killed
     # Check that background thread has terminated
-    for id, thread in threading._active.items():
-        assert thread.name != "GoServerBackgroundThread"
+    monitor_thread = None
+    for thread in threading.enumerate():
+        if thread.name == "GoServerMonitorThread":
+            monitor_thread = thread
+    assert monitor_thread
+    assert monitor_thread.is_alive()
+    assert monitor_thread.isDaemon()
```

Comment:
same as previously, I believe you can just use `daemon` as a property

Reply:
yup